### PR TITLE
Unit test files consistency

### LINKED ISSUE #925: Inconsistent naming of unit test files
## Issue

The python source code files that contain unit tests have inconsistent names. Some have names like `test_*.py` (e.g. `monkey/common/network/test_network_utils.py`) while others have names like `*_test.py` (e.g. `infection_monkey/exploit/tools/payload_parsing_test.py`). To further complicate matters, some files, such as `monkey_island/cc/resources/test/monkey_test.py`, are not unit tests at all.

## Solution

Both [pytest](https://docs.pytest.org/en/reorganize-docs/new-docs/user/naming_conventions.html) and [unittest](https://docs.python.org/3/library/unittest.html#basic-example) use the naming convention `test_*.py`. Rename all of the files with names like `*_test.py` that contain unit tests so that they are consistent with pytest and unittest recommendation.

Where applicable, the class names should match the filenames. For example, the file containing `class TestBootloader` should be named `test_bootloader.py`

- [x] All files containing unit tests conform to the naming convention `test_*.py`
- [x] Class names match file names
- [x] All pre-existing unit tests run and pass
- [x] Refactoring/renaming does not introduce any bugs or break any functionality

Time Remaing -  0.25d - @shreyamalviya

### PR DESCRIPTION
**Pull request by shreyamalviya**
# What does this PR do? 

Fixes #925 

## PR Checklist
* [x] Have you added an explanation of what your changes do and why you'd like to include them?
* [ ] Is the TravisCI build passing? 
* [ ] Was the CHANGELOG.md updated to reflect the changes?
* [ ] ~Was the documentation framework updated to reflect the changes?~ NA

## Testing Checklist

* [ ] ~Added relevant unit tests?~ NA
* [x] Have you successfully tested your changes locally? Elaborate:
    - Existing unit tests pass 
* [ ] If applicable, add screenshots or log transcripts of the feature working

### CODE REVIEW

**Review by mssalvatore — CHANGES_REQUESTED** [review]

File: monkey/monkey_island/cc/resources/test/monkey_test_endpoint.py [hunk 1]
```diff
@@ -6,7 +6,7 @@
 from monkey_island.cc.resources.auth.auth import jwt_required
 
 
-class MonkeyTest(flask_restful.Resource):
+class MonkeyTestEndpoint(flask_restful.Resource):
```
Comment (VakarisZ) [inline monkey/monkey_island/cc/resources/test/monkey_test_endpoint.py:9]: Term `test` is still confusing. If possible, I'd rename the path to `monkey_island/cc/resources/blackbox/blackbox_monkey`. When you see the term blackbox it obvious what it means, meanwhile `test` doesn't tell a whole lot and also confuse you between unit test or different kind of test.

**Review by VakarisZ — CHANGES_REQUESTED** [review]
Would be ideal if you could fix the naming of blackbox resources by replacing word `test` with word `blackbox` where it makes sense. This includes the URI's of the resources

File: monkey/monkey_island/cc/app.py [hunk 2]
```diff
@@ -12,6 +12,10 @@
 from monkey_island.cc.resources.attack.attack_report import AttackReport
 from monkey_island.cc.resources.auth.auth import Authenticate, init_jwt
 from monkey_island.cc.resources.auth.registration import Registration
+from monkey_island.cc.resources.blackbox.clear_caches import ClearCaches
+from monkey_island.cc.resources.blackbox.log_blackbox_endpoint import LogTestEndpoint
+from monkey_island.cc.resources.blackbox.monkey_blackbox_endpoint import MonkeyTestEndpoint
+from monkey_island.cc.resources.blackbox.telemetry_blackbox_endpoint import TelemetryTestEndpoint
```
Comment (VakarisZ) [inline monkey/monkey_island/cc/app.py:18]: Rename classes to match filenames

**Review by VakarisZ — CHANGES_REQUESTED** [review]
One small thing remaining: rename classes to match filenames. Class in `monkey_island.cc.resources.blackbox.log_blackbox_endpoint` should be named `LogBlackboxEndpoint`, not `LogTestEndpoint`

**Review by mssalvatore — APPROVED** [review]

**Pull request merged**